Picha review

wmahad opened the pull request:
this is an application where a user can upload a picture, customise it with a few effects and then share the picture on his wall.

wmahad committed:
(feature) Create API models

- Add UserPhoto models
- Add Effects models
- Add API migrations

wmahad committed:
(chore) Add tests for models

- Add test_models.py

wmahad committed:
(chore) Add App Mockups

- Add UI mockups
- Add requirements.txt file

wmahad committed:
(feature) Add model serializers

- Add api permissions
- Add api urls
- Add api model serializers

wmahad committed:
(chore) Add tests for views

- Add tests for api views

wmahad committed:
(feature) Add image transformation methods

- Add api/image_effects.py
- Add api/utilities.py

wmahad committed:
(feature) Add API views

- Add api/views.py

wmahad committed:
(feature) Add UI templates

- Add landing template
- Add dashboard template
- Add filters template
- Add navbar template
- Add textbox template

wmahad committed:
(chore) Add Media Files

- Add site files
- Add AngularJs scripts

wmahad committed:
(chore) Add media directory

- this directory is where the uploaded
- images will be stored in

wmahad committed:
(fix) Fix Heroku Configs

- Add Procfile
- Update requirements.txt
- Update README.md

wmahad committed:
(fix) Heroku Configs

wmahad committed:
(fix) Import error

wmahad committed:
(fix) File deletion

wmahad committed:
(fix) python anti-patterns

wmahad committed:
(fix) python anti-patterns and README.md

andela-anandaa commented on api/views.py:
```diff
@@ -0,0 +1,312 @@
+import urllib, cStringIO
```
**Code Quality - Nitpick**
Should be on separate lines.

andela-anandaa commented:
#### README and Documentation
- Fix minor issues on the PROJECTPLAN.md
- Title: "Picha: a Django photo editing application"
- Capitalization of proper nouns: _Facebook, Python, _
- Titles and Sub-titles should be written in _Title Case_.
- Add QuantifiedCode badge

#### Git Commits
- Fix inconsistencies in the commit messages

#### Commenting / Code Documentation
- All methods are well commented and the code is self-documenting

#### Code Quality
- Code quality is satisfactory, from [QuantifiedCode analysis](https://www.quantifiedcode.com/app/project/a57180184fe445b982f1097b4b970bcb?branch=origin%2Fdevelop)

#### Testing
- 66% code coverage

#### Functionality
- Excellent: added a lot of other functionality/effects.
- Used DRF, making the app fully RESTful (exceeding expectations)
- Fix FB login issue (development mode)

#### UI/UX
- Very minor issues
- Further reading: https://www.nngroup.com/articles/ten-usability-heuristics/, http://architectingusability.com/2012/06/28/donald-normans-design-principles-for-usability/

wmahad committed:
(fix) Fix Issues From Review

Following are changes made to README.md:
- Changed the title to Picha: a Django photo editing application
- Capitalized all proper nouns e.g. Facebook
- Changed all headings and sub-headings to 'Title Case' styles.
- Added QuantifiedCode badge

Changes made to api/views.py include:
- Fixed the imports to two separate lines
- Commented out code not being used now, it's ideally meant to work in version two of the application

Others changes made include:
- Added a coveralls ignore file to restrict coveralls to what it should track when testing the application
- Changed PROJECTPLAN.md file emphasis formats to bold
- Added tests for image effects i.e. filters, rotations, text addition and enhancements- Fixed title's overflow property of the filter panel to fixed

wmahad committed:
(fix) fix import error

Remove ipdb import from views.py which was causing bdbquit exception on heroku

wmahad committed:
(fix) fix file deletion

Add try... except.. block to handle safe deletion of files when a delete command is issued

wmahad committed:
(fix) fix python anti-patterns

I changed the variable names to pep8 compliant names plus removed wildcard imports

wmahad committed:
(fix) fix brightness effect

when a user selects does not select any of the components under the enhancement panel the effect is not applied on to the image.

wmahad committed:
(fix) fix travis error

travis is failing to install the required nose module. Add django_nose and nose module to the requirements.txt file

wmahad committed:
(fix) Update README.md

Remove full stop in title and place badges on the next line.
